paasta_tools/cli/utils.py

```diff
@@ -660,15 +662,17 @@ def get_jenkins_build_output_url():
     return build_output
 
 
-def get_instance_config(service, instance, cluster, soa_dir, load_deployments=False):
+def get_instance_config(service, instance, cluster, soa_dir, load_deployments=False, instance_type=None):
     """ Returns the InstanceConfig object for whatever type of instance
     it is. (chronos or marathon) """
-    instance_type = validate_service_instance(
-        service=service,
-        instance=instance,
-        cluster=cluster,
-        soa_dir=soa_dir,
-    )
+    if instance_type is None:
```

Comment:
I promise this is a good thing.

Now we can actually speed things up by "injecting" what it is we are looking for instead of searching the filesystem each time.